src/synthesizer/emission_models/stellar/models.py

```diff
@@ -260,41 +307,43 @@ def __init__(
             label (str): The label for this emission model.
             fesc (float): The escape fraction of the emission.
             feac_ly_alpha (float): The escape fraction of Lyman-alpha.
-            linecont (EmissionModel): The line continuum model to use, if None
-                then one will be created. Only used if fesc_ly_alpha < 1.0.
+            nebular_line (EmissionModel): The nebular line model to use, if
+                None then one will be created. Only used if
+                fesc_ly_alpha < 1.0.
             nebular_continuum (EmissionModel): The nebular continuum model to
                 use, if None then one will be created. Only used if
                 fesc_ly_alpha < 1.0.
         """
-        # If we have a Lyman-alpha escape fraction then we need to combine
-        # the Lyman-alpha line emission with the nebular continuum emission
-        # and the rest of the line emission
-        if fesc_ly_alpha < 1.0:
-            # Make a line continuum model if we need one
-            if linecont is None:
-                linecont = LineContinuumEmission(
-                    grid=grid,
-                    fesc=fesc_ly_alpha,
-                    **kwargs,
-                )
-
-            # Make a nebular continuum model if we need one
-            if nebular_continuum is None:
-                nebular_continuum = NebularContinuumEmission(
-                    grid=grid,
-                    fesc=fesc,
-                    **kwargs,
-                )
+        # Make a nebular line model if we need one
+        if nebular_line is None:
+            nebular_line = NebularLineEmission(
+                grid=grid,
+                fesc_ly_alpha=fesc_ly_alpha,
+                fesc=fesc,
+                **kwargs,
+            )
 
+        # Make a nebular continuum model if we need one
+        if nebular_continuum is None:
+            nebular_continuum = NebularContinuumEmission(
+                grid=grid,
+                fesc=fesc,
+                **kwargs,
+            )
+
+        # If we have a Lyman-alpha escape fraction then calculate the
+        # updated line emission and combine with the nebular continuum.
+        if fesc_ly_alpha < 1.0:
```

Comment:
Alternative to the above: Nebular is always a combination of the lines and continuum and we remove this logic here?

Reply:
The logic here was that just grabbing an array from the grid will be quicker, but I guess we're grabbing the other arrays already. 

As an aside, a nice addition to the test suite would be something that grabs these from the emission model and comparea to the `nebular` key from the grid.